Framework/Geometry/src/Instrument/ReferenceFrame.cpp

```diff
@@ -157,5 +157,23 @@ const V3D ReferenceFrame::vecPointingAlongBeam() const {
   return m_vecPointingAlongBeam;
 }
 
+/**
+Convenience method for checking whether or not a vector is aligned
+with the along beam vector.
+@param v vector to be tested against along beam vector
+@return result of whther the along beam and test vector are parallel.
+*/
+bool ReferenceFrame::isVectorPointingAlongBeam(const V3D &v) const {
+  V3D vec = v;
+  vec.normalize();
+  bool isPointing = false;
```

Comment:
All of this [section](isPointing) could have been collapsed down into a single line:
```cpp
return m_vecPointingAlongBeam.scalar_prod(vec) == 1;
```
Only worth fixing if you have other changes to make I think.